dask/array/core.py

```diff
@@ -466,6 +466,35 @@ def squeeze(a, axis=None):
     return Array(dsk, b.name, chunks, dtype=a.dtype)
 
 
+def topk(k, x):
+    """ The top k elements of an array
+
+    Returns the k greatest elements of the array in sorted order.  Only works
+    on arrays of a single dimension.
+
+    >>> x = np.array([5, 1, 3, 6])
+    >>> d = from_array(x, chunks=2)
+    >>> d.topk(2).compute()
+    array([6, 5])
+
+    Runs in near constant time, returns all results in a single chunk so
```

Comment:
Maybe you mean "run in near *linear* time"? Given that it looks at every array element, I'm pretty sure it's not constant time....